# RNAseq data analysis

## Description

The output data (FASTQ files) are mapped to the target genome using the rnaStar aligner [Dobin2012] that maps reads genomically and resolves reads across splice junctions. We use the 2 pass mapping method outlined in [Engström2013] in which the reads are mapped twice. The first mapping pass uses a list of known annotated junctions from Ensemble. Novel junctions found in the first pass are then added to the known junctions and a second mapping pass is done (on the second pass the RemoveNoncanoncial flag is used). After mapping we post process the output SAM files using the PICARD tools to: add read groups, AddOrReplaceReadGroups which in additional sorts the file and coverts it to the compressed BAM format.

We then compute the expression count matrix from the mapped reads using HTSeq (www-huber.embl.de/users/anders/HTSeq) and one of several possible gene model databases. The raw count matrix generated by HTSeq are then be processed using the R/Bioconductor package DESeq (www-huber.embl.de/users/anders/DESeq) which is used to both normalize the full dataset and analyze differential expression between sample groups.

The data was clustered using the normalized counts of all genes. In additional a multi-dimensional scaling analysis was done also using the normalized counts of all genes. Finally, a heatmap was generated using the heatmap.2 function from the gplots R package. The data plot was the mean centered normalized log2 expression of the top 100 significant genes. For simple hierarchical clustering the correlation metric was used (Dij = 1 - cor(Xi,Xj)) with the Pearson correlation on the normalized log2 expression values.

## Program versions

| Program | Version |
|---------|---------|
|HTSEQ | htseq/HTSeq-0.5.3
|PICARD | picard/picard-tools-1.124
|R | R/R-3.2.0
|STAR | star/STAR-STAR_2.5.0a
|SAMTOOLS | samtools/samtools-0.1.19


## Data Files

- Human:
	- GENOME:		UCSC HG19
	- GTF: 		gencode.v18.annotation

- Mouse:
	- GENOME:		UCSC MM10
	- GTF: 		Mus_musculus.GRCm38.80

## References:

[Dobin2012] A. Dobin et al, _STAR: ultrafast universal RNA-seq aligner._ Bioinformatics 2012.

[Engström2013] Engström, et. al., _Systematic evaluation of spliced alignment programs for RNA-seq data_, Nat. Meth., **10**, pp. 1185, 2013.